### PR TITLE
Update FxPool deploy script to use FxPoolFactory

### PR DESCRIPTION
**Pull request by schystz**
## Description of Changes

Updated `deployFxPool.ts` with the ff changes:
- use FxPoolFactory to deploy an FxPool
- new step: set collector address of FXPool to deployer
- use fast gas to avoid deploy errors on kovan
- verify FxPool on etherscan

## How To Test

`yarn deploy:fx-pool`

1. select kovan
2. choose any base token
3. choose any name, symbol and fee (e.g. XLP, XLP, 9)
4. fresh deploy
  - YES: everything will be deployed
  - NO: only the FxPool and assimilator (if not existing) will be deployed, libs and factory will be reused
    - refer [here](https://github.com/xave-finance/halodao-contract-addresses/pull/25) for the existing kovan deployment

### Developer Checklist:

- [x] I have followed the guidelines in our Contributing document
- [ ] This PR has a corresponding JIRA ticket
- [ ] My branch conforms with our naming convention i.e. `feature/HDEV-XXX-description`
- [ ] I have written new tests for your core changes, as applicable
- [ ] I have successfully ran tests locally
- [ ] I have formatted my code using format document in VSCode

### Reviewers Checklist:

- [ ] Code is readable and understandable; any unclear parts have explanations
- [ ] I have successfully ran tests locally

### CODE REVIEW

File: scripts/pool-actions/deployFxPool.ts [hunk 1]
```diff
@@ -23,9 +26,15 @@ export default async (taskArgs: any) => {
   const network = taskArgs.to
   const baseToken = taskArgs.basetoken
   const freshDeploy = taskArgs.fresh ? taskArgs.fresh === 'true' : false
+  const name = taskArgs.name
+  const symbol = taskArgs.symbol
+  const fee = taskArgs.fee
```
Comment (mystbrent) [inline scripts/pool-actions/deployFxPool.ts:31]: just a suggestion maybe its good to be verbose on what gas unit is this in? I'm assuming this is using gwei as unit?

File: scripts/cli/deployFxPoolCLI.ts [hunk 2]
```diff
@@ -20,16 +19,38 @@ inquirer
...
+    },
+    {
+      type: 'input',
+      name: 'symbol',
+      message: 'LP token symbol',
+    },
+    {
+      type: 'input',
+      name: 'fee',
+      message: 'Protocol fee (ETH units)',
```
Comment (mystbrent) [inline scripts/cli/deployFxPoolCLI.ts:35]: wei unit?

**Review by mystbrent — APPROVED** [review]
Everything looks good thanks for pushing changes @schystz !

One minor thing though, I noticed there is an old file under pool-actions directory - https://github.com/xave-finance/amm-contracts/blob/feature/deploy-script-update/scripts/pool-actions/deployPooll.ts

might be good to delete it :)

File: scripts/cli/deployFxPoolCLI.ts (hunk 2 above)
Comment (schystz) [reply scripts/cli/deployFxPoolCLI.ts:35]: Ether! :) 

We use `parseUnits()` before feeding this to the contract

File: scripts/pool-actions/deployFxPool.ts (hunk 1 above)
Comment (schystz) [reply scripts/pool-actions/deployFxPool.ts:31]: We indicate it on the CLI script:
https://github.com/xave-finance/amm-contracts/blob/e6eb5c4fe9c200e39ab7f5d180a6e0ea6382503b/scripts/cli/deployFxPoolCLI.ts#L35

Or @mystbrent do you mean adding a comment here?

**Comment by 0xAplki** [comment]
will check this after we merge https://github.com/xave-finance/amm-contracts/pull/20

**Review by 0xAplki — APPROVED** [review]

**Pull request merged**